### DISCUSSION TITLE
实现一个 solana 链上的 casbin loader

### DISCUSSION THREAD

**Discussion by v1xingyue** [creatorsand/.github#7] [Ideas]
[casbin](https://casbin.org/) 是一个通用的权限解决方案。
但是，依然是一个基于传统数据存储的解决方案。 

项目通过将权限数据存储在solana 链上，以达到权限控制的无中心化。
从而实现，去中心化的开源软件分发机制。

应用暂定使用 native 合约开发，基于 wallet adapter 实现简单的应用包装。

应用分为两个部分: 

1. solana 链上的权限方案数据存储
2. 用rust 一个solana 链上的 loader 
3. 应用 2 中的的loader 完成应用的权限控制

**Comment by DaviRain-Su** [comment]
https://build.superteam.fun/ideas/sign-in-with-solana 

reference: 
- https://twitter.com/phantom/status/1693759783512543368
- https://oauth.net/2/
- https://siws.web3auth.io/
- https://phantom.app/learn/developers/sign-in-with-solana

**Reply by DaviRain-Su** [reply]
"Sign-in with Solana"是一种简化Solana生态系统中用户登录体验的解决方案。它类似于OAuth2的登录方式，允许用户通过一键登录到Solana网络上的各种应用和产品。

开发者可以创建一个授权服务器来验证用户对Solana钱包的访问权限，并发送访问令牌以进行登录。他们还可以构建一个开放的API，供其他应用集成"Sign-in with Solana"按钮。

"Sign-in with Solana"使用Web3Auth协议，它是一个用于在Web应用中进行去中心化身份验证的开放标准。它允许用户使用他们的Solana钱包来进行登录，而无需输入用户名和密码。用户可以选择使用他们喜欢的Solana钱包（如Phantom、Sollet等）来进行登录，并授权应用访问他们的钱包信息。

通过"Sign-in with Solana"，用户可以更方便地访问和使用Solana生态系统中的各种应用和产品，无需重复输入钱包地址或私钥。
